### PR TITLE
Images search by text

### PR DESCRIPTION
**Pull request by gusartus**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
     "react-masonry-css": "1.0.12",
     "react-redux": "6.0.0",
     "react-router-dom": "4.3.1",
+    "react-string-replace": "^0.4.1",
```
Comment (CodeAnnihilator) [inline package.json]: test

**Review by CodeAnnihilator — CHANGES_REQUESTED** [review]
tesolve test

File: package.json (hunk 1 above)
Comment (Gerasimov94) [inline package.json]: remove ^

File: src/common/selectors/entities.ts [hunk 2]
```diff
@@ -10,6 +11,24 @@ export const getBuildingImages = createSelector(
 	images => images.filter((item: any) => item.build.status === 'building'),
 );
 
+export const getFilteredImages = createSelector(
+	getImages,
+	getImagesTextFilter,
+	(images, substr) => {
+		if (images.length && substr) {
+			return images.filter((image: any) => {
+				if (
```
Comment (Gerasimov94) [inline src/common/selectors/entities.ts:20]: filter works with predicates, you don't need to use extra if's. Secondly, use a String.prototype.includes() instead image.distro.full_name.toLowerCase().indexOf(substr.toLowerCase()) !== -1

File: src/modules/Dashboard/actions/dashboard.ts [hunk 3]
```diff
@@ -1,10 +1,15 @@
 import {createAction} from 'redux-actions';
 import types from '../constants/dashboard';
 
-export const selectImage = (id: string | number) => ({
-	type: types.SELECT_IMAGE,
-	payload: id,
-});
+export const selectImage = createAction(
+	types.SELECT_IMAGE,
+	(id: string | number) => ({id}),
```
Comment (Gerasimov94) [inline src/modules/Dashboard/actions/dashboard.ts:6]: id: ReactText

File: src/modules/Dashboard/actions/dashboard.ts [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
+);
+
+export const setImagesTextFilter = createAction(
+	types.UPDATE_TEXT_FILTER,
+	(text: any) => ({text}),
```
Comment (Gerasimov94) [inline src/modules/Dashboard/actions/dashboard.ts:11]: text === any? maybe ReactText enough for description?

File: src/modules/Dashboard/components/ImagesCardView/ImageCard/ImageCard.tsx [hunk 5]
```diff
@@ -48,8 +50,21 @@ export default class ImageCard extends Component<IProps, IState> {
 		image: {
 			thumb: '',
 		},
+		textFilter: '',
 	};
 
+	public findInTextWrapper = (text: string) => {
+		if (this.props.textFilter) {
+			return (
+				reactStringReplace(text, this.props.textFilter, (match: string, i: number) => (
```
Comment (Gerasimov94) [inline src/modules/Dashboard/components/ImagesCardView/ImageCard/ImageCard.tsx:59]: https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/String/replace

File: src/modules/Dashboard/components/ImagesCardView/ImageCard/imageCard.scss [hunk 6]
```diff
@@ -232,6 +232,13 @@ $note-line-height: 18px;
 	justify-content: center;
 }
 
+.highlighted {
+	background-color: rgb(32, 183, 51);
+	color: #FFF;
```
Comment (Gerasimov94) [inline src/modules/Dashboard/components/ImagesCardView/ImageCard/imageCard.scss:237]: use a contant

File: src/modules/Dashboard/components/ImagesCardView/ImageCard/imageCard.scss [hunk 7]
```diff
@@ -232,6 +232,13 @@ $note-line-height: 18px;
 	justify-content: center;
 }
 
+.highlighted {
+	background-color: rgb(32, 183, 51);
```
Comment (Gerasimov94) [inline src/modules/Dashboard/components/ImagesCardView/ImageCard/imageCard.scss:236]: here a rgb, lower a hex. in cusdeb we are use hex

File: src/modules/Dashboard/reducers/dashboard.ts [hunk 8]
```diff
@@ -6,16 +6,26 @@ const initialState = {
...
 		...state,
-		activeImageId: action.payload,
+		activeImageId: action.payload.id,
 		isPending: false,
 	}),
 	[types.SET_ACTIVE_IMAGES_STATUS]: (state: any , action: any) => ({
 		...state,
 		activeImagesStatus: action.payload,
 	}),
+	[types.UPDATE_TEXT_FILTER]: (state: any , action: any) => ({
```
Comment (Gerasimov94) [inline src/modules/Dashboard/reducers/dashboard.ts:24]: {payload: {text}

File: src/modules/Dashboard/reducers/dashboard.ts [hunk 9]
```diff
@@ -6,16 +6,26 @@ const initialState = {
...
+		...state,
+		displaying: {
+			...state.displaying,
+			textFilter: action.payload.text,
```
Comment (Gerasimov94) [inline src/modules/Dashboard/reducers/dashboard.ts:28]: textFilter: text, or change param name in action, and write:

displaying: {
			...state.displaying,
			textFilter,
}

this is much cleaner then you write

File: src/modules/ImageCreation/reducers/imageCreationInitializationReducer.ts [hunk 10]
```diff
@@ -59,15 +59,15 @@ export default handleActions({
 	[types.SELECT_ENTITY]: (state: any , {payload: {entity, type, isSelected}}) => {
 		let isFounded = false;
 		let nextIndex = parseInt(state.currentInitializationSlide, 10);
-		const entitesFilters = {...state.entitesFilters};
-		const filtersTypes = Object.keys(entitesFilters);
+		const entitiesFilters = {...state.entitiesFilters};
```
Comment (Gerasimov94) [inline src/modules/ImageCreation/reducers/imageCreationInitializationReducer.ts:62]: what the mess inside that function? talk with Eugene, he helps you to find a better solution for this.

File: src/modules/ImageCreation/selectors/imageCreationSelectors.tsx [hunk 11]
```diff
@@ -84,10 +84,10 @@ export const getAlertTitleKey = createSelector(
 	},
 );
 
-export const makeGetInputValue = () => (state: any, type: string) => getEntitesFilters(state)[type];
+export const getInputValue = () => (state: any, type: string) => getEntitiesFilters(state)[type];
 
-export const makeGetImagesByType = () => createSelector(
-	getDistros, getTargetDevices, getBuildTypes, getBrands, makeGetInputValue(),
+export const getImagesByType = () => createSelector(
```
Comment (Gerasimov94) [inline src/modules/ImageCreation/selectors/imageCreationSelectors.tsx:89]: this is creator of createSelector. add a make before that.

**Review by Gerasimov94 — CHANGES_REQUESTED** [review]

**Comment by CodeAnnihilator** [comment]
change commit message according to our standards

**Review by CodeAnnihilator — CHANGES_REQUESTED** [review]
change commit message according to our standards

**Pull request closed without merging**